### PR TITLE
Update SDK documentation for pairing code based auth

### PR DESCRIPTION
**Pull request by henrikjonhed**
Fixes PLC-879

Updates the SDK documentation in general and specifically for promoting the use of paring code based auth over hard-pairing.

### CODE REVIEW

File: sdk/index.md [hunk 1]
```diff
@@ -15,171 +15,101 @@
 
 ## Introduction
 
-The Soundtrack SDK enables you to build support for Soundtrack on your platform. By combining expert curation and world-class tech, Soundtrack Your Brand provides a beautiful all-in-one solution for streaming music to stores, hotels, restaurants, and other commercial settings.
+The Soundtrack SDK makes it possible to play music from Soundtrack on your platform. 
+It is available for selected partners with potential for over 5000 locations. 
+If you're interested in becoming one, please reach out to [sdk@soundtrackyourbrand.com](mailto:sdk@soundtrackyourbrand.com).
 
-The SDK is available for selected partners. If you're interested in becoming one, please reach out to sdk@soundtrackyourbrand.com (applications currently open for opportunities of more than 5000 locations).
+All applications using the SDK must adhere to the Soundtrack Terms & Conditions. Certification is required before launch.
```
Comment (FWennerdahl) [inline sdk/index.md:22]: Should we also add
`Certification is required before launch and following any non-trivial changes in the integration.`
or something similar?

File: sdk/index.md [hunk 2]
```diff
@@ -15,171 +15,101 @@
...
 
-This page contains the documentation for the SDK. In order to use it, you will need our latest build (which we will send to you).
+### API
+The SDK provides C header files, exposing an API called SPlayer API. It enables you to can do basic playback control and interface to the audio output of your choice.
```
Comment (FWennerdahl) [inline sdk/index.md:25]:
```suggestion
The SDK provides C header files, exposing an API called SPlayer API. It enables you to do basic playback control and interface to the audio output of your choice.
```

File: sdk/index.md (hunk 1 above)
Comment (henrikjonhed) [reply sdk/index.md:22]: Thats a very good point given previous experiences. Would you like to write a suggestion?

Comment (FWennerdahl) [reply sdk/index.md:22]:
```suggestion
All applications using the SDK must adhere to the Soundtrack Terms & Conditions. Certification of the integration is required before launch. Any non-trivial modification to the integration post-launch requires re-certification before it's released.
```

File: sdk/index.md [hunk 3]
```diff
@@ -15,171 +15,101 @@
...
 
-5. Run the query and you should get a pairing code as output. You will need this code to pair the created device with a specific sound zone later.
-6. Write down the hardware_id that you chose (in the example ‘partner_id_ab12’). Write down the pairing code that was output when you ran the query. You can rerun the query if you need to. Supplying an existing hardware id will always return the same pairing code.
+1. Your application initiates a redirect to the following URL:
+   `http://business.soundtrackyourbrand.com/connect/generic?redirect=<REDIRECT_URL>`
+   Replace `<REDIRECT_URL>` with the URL of your application that will handle the callback. This URL must be accessible by the device that is used for the pairing flow.
+1. The user logs in to their Soundtrack account and selects the desired sound zone.
+1. Soundtrack redirects the user back to your specified REDIRECT_URL with the pairing code appended as a query parameter: `<REDIRECT_URL>?code=<CODE>`
+1. Your application can then extract the code parameter and programmatically provide it to the SDK. See `example_auth.c` in the SDK package for a proof-of-concept using this method.
```
Comment (FWennerdahl) [inline sdk/index.md:91]: Many step 1.

File: sdk/index.md [hunk 4]
```diff
@@ -15,171 +15,101 @@
...
- alsa_audio_impl::audio_pause: 0
-ERROR: EBADFD: Let's prepare alsa again
-snd_pcm_prepare returned 0
-```
+## Building and running the example code
+1. Open the /src folder. There are several examples showcasing some basic setup of a player application.
+1. Build the project using cmake
+1. There will now be several example targets in the build root folder
+1. Run the example executable (eg. libsplayer_simple). When running an example for the first time, provide the pairing code (see above) as the first argument to the application.
+1. There should be some console output and music should start playing shortly
```
Comment (FWennerdahl) [inline sdk/index.md:109]: Here too.

File: sdk/index.md (hunk 3 above)
Comment (henrikjonhed) [reply sdk/index.md:91]: https://www.markdownguide.org/basic-syntax/#ordered-lists

File: sdk/index.md [hunk 5]
```diff
@@ -293,29 +228,29 @@ Alsa initialized, buffer_size: 88200 period_size: 22050
...
-ERROR_CHANNEL_ASSIGNED | Checks if any music is assigned | Fails if no music assigned
+ERROR_ACTIVE_SUBSCRIPTION | Checks if player have an active subscription | Fails if inactive.
+ERROR_CHANNEL_ASSIGNED | Checks if any source of music is assigned | Fails if no music is assigned
 ERROR_DOWNLOADED_DATA | Checks downloaded data for the assigned music. | Fails if nothing is downloaded.
-ERROR_NO_VOLUME | Checks the volume level | Fails if there is no volume, level is zero.
-ERROR_DISKCACHE_LOW | Checks free disk space | Fails if free disk space is below 1 GB.
-ERROR_DISKCACHE_CRITICALLY_LOW | Checks free disk space | Fails if free disk space is below 256 MB.
-ERROR_PAYMENT_EXPIRED | Check payment status | Fails if payment has expired.
-ERROR_CLOCK_WRONG | Check server time offset | Fails if server time offset differs +- 15 min
+ERROR_NO_VOLUME | Checks the volume level | Fails if there is volume is set to zero.
```
Comment (FWennerdahl) [inline sdk/index.md:249]:
```suggestion
ERROR_NO_VOLUME | Checks the volume level | Fails if the volume is set to zero.
```

File: sdk/index.md [hunk 6]
```diff
@@ -293,29 +228,29 @@ Alsa initialized, buffer_size: 88200 period_size: 22050
...
+ERROR_DISKCACHE_LOW | Checks free disk space | Fails if the free disk space is below 1 GB.
+ERROR_DISKCACHE_CRITICALLY_LOW | Checks free disk space | Fails if the free disk space is below 256 MB.
+ERROR_PAYMENT_EXPIRED | Check payment status | Fails if the paired zone doesn't have a valid and payed subscription.
```
Comment (FWennerdahl) [inline sdk/index.md:252]:
```suggestion
ERROR_PAYMENT_EXPIRED | Check payment status | Fails if the paired zone doesn't have a valid and paid subscription.
```

**Review by FWennerdahl — APPROVED** [review]
Fix the typos above and we should be good to go 👍

**Pull request merged**